### DISCUSSION TITLE
Modify Openapi output with custom annotation?

### DISCUSSION THREAD

**Discussion by HerrDerb** [smallrye/smallrye-open-api#2084] [Q&A]
Because of backwards compatibility, we have to keep some api endpoints for a while longer. Currently we mark them with `@Deprecated` and `@Schema(hidden =true)`
This results in a lot of deprecated warnings and also prone to forgetting the hidden annotation.

As we have to support those endpoints still quite a while, I would like to use a custom annotation for this. (better fields for descriptions and not deprecated warnings)

I would like to achive, that a field or an operation or a schema annotated with `@MyCustomDeprecation` gets marked as deprecated too in the generated openapi scheme.

I had had a look with `OSAFilter` but there, I don't have any knowledge anymore about the annotations on the objects. Also I tried to add `@Tag` to my custom annotation in hope, that I could use this tag in the filter do modify the openapi output. But it seems that the `@Tag` on a annotation is not applied.

Also `@Schema(hidden=true)` is not applicable on an annotation type.

Is there a way I can influence the openapi output with a custom annotation? Specifically, set an operation, field or schema as hidden?

**Comment by MikeEdgar** [comment]
You could try using an `@Extension`  annotation with a value you can find in your OASFilter implementation. Then you can modify the operation or schema as needed. The extension can be removed or left alone when your filter runs.

**Comment by HerrDerb** [comment]
Hm could not really make it work like that. I now did implement an annotation processor that checks that hidden is present. This does the job
